### PR TITLE
Make `List` documentation more consistent

### PR DESCRIPTION
**Pull request by shardulc**
I noticed some inconsistencies in the language used in the documentation of [`List`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html), some of which are inherited from elsewhere. This is a pretty minor issue but I thought I'd just make sure it's known.

### Part of speech of docstring headline

It seems like the rule is that a docstring headline should be either:
+ A noun phrase, in which case we assume that the method evaluates to the object being described. E.g. [`length`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#length-0): "The length (number of elements) of the list."
+ "Alias for <some other method>." E.g. [`++`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#++-1dd): "Alias for concat"
+ A verb conjugated in the third-person singular. E.g. [`isEmpty`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#isEmpty-0): "Tests whether the list is empty."

However, this rule is broken in the following instances:
+ Incorrect verb conjugations (they are all imperative): [`foreach`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#foreach-58c), [`apply`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#apply-aba), [`copyToArray`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#copyToArray-fffffd03), [`search`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#search-aaf), [`to`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#to-fffff006), [`toArray`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#toArray-fffff5df), [`hashCode`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#hashCode-ffffff27).
+ Noun phrase that doesn't describe the output of the method, but comments on the method itself: [`equals`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#equals-4ad), [`canEqual`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#canEqual-4ad).
+ Something else: [`sameElements`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#sameElements-356) phrased as a question.

I tried to fix all of the above in b2634acb, except for `equals` & `canEquals` and `sameElements` which I'm less sure about, so those are in separate commits 2fb96465 and 72567e11 respectively.

### `takeWhile` defined as "taking" elements

[`take`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#take-fffff84b) & [`drop`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#drop-399) are described as "selecting" certain elements, whereas [`takeWhile`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#takeWhile-5a1) and [`dropWhile`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#dropWhile-883) are described as "taking" or "dropping" certain elements. Is there a reason for this difference? Why not describe the latter as "selecting" too? Do they actually build/depend upon the functionality of `take` and `drop`?

As a proposal, I changed all of their docstrings to say "select" in 472497df.

### Missing documentation

The following methods are missing documentation:
+ [`slice`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#slice-fffff472)
+ [`toList`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#toList-0), [`toBuffer`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#toBuffer-fffff87f), & kin.
+ [`iterator`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#iterator-0) (the docstring shown is not really documentation)
+ [`knownSize`](https://www.scala-lang.org/api/3.3.1/scala/collection/immutable/List.html#knownSize-0)

I added proposed documentation in 27a1146e, 1f464e58, 64386ab7, aaee7195.

### Formatting and grammatical errors

Proposed fixes in f7ae2f1b.

### Related PR to `dotty`

In one instance, the docstring that needed to be changed was in the Scala 3 library, not Scala 2, so I made a [tiny PR there](https://github.com/lampepfl/dotty/pull/19108).

## Summary

It may look like these changes touch a lot of different files arbitrarily but there is a unifying principle: I just wanted the documentation of `List` to be consistent. The backstory is that in a programming course I help teach, we give students a copy of parts of the `List` documentation in their reference sheets for an exam, and we always strive to make all the material we give them in an exam absolutely error-free in both code and prose.

Thanks for your attention and I am happy to work with someone more experienced to refine these changes if needed!

### CODE REVIEW

File: src/library/scala/collection/IterableOnce.scala [hunk 1]
```diff
@@ -42,7 +42,13 @@ import scala.runtime.{AbstractFunction1, AbstractFunction2}
   * @define coll collection
   */
 trait IterableOnce[+A] extends Any {
-  /** Iterator can be used only once */
+
+  /** An [[scala.collection.Iterator]] over the elements of this collection.
```
Comment (lrytz) [inline src/library/scala/collection/IterableOnce.scala]: Use `$coll` (also in `toList` / `toVector` etc below)?

**Review by lrytz — COMMENTED** [review]
LGTM otherwise, thank you!

Could you squash everything into one commit?

**Comment by shardulc** [comment]
Done, thanks for the review!

**Comment by som-snytt** [comment]
The other PR for doc is https://github.com/scala/scala/pull/10605. Possibly there is no overlap or collision.

I started working on a quickie option for scaladoc to warn about missing doc, but I haven't refined it yet. I see it ought to detect missing text, exclusive of example blocks and other elements.

I was surprised that `slice` isn't over-documented, as it received extra attention over time. `List` is also my go-to for paradigms.

**Review by ansvonwa — APPROVED** [review]
LGTM
Thank you for improving the docs!

**Review by jxnu-liguobin — APPROVED** [review]

**Pull request merged**